libraries/botbuilder-dialogs-adaptive/src/adaptiveDialog.ts

```diff
@@ -1038,165 +1027,314 @@ export class AdaptiveDialog<O extends object = {}> extends DialogContainer<O> im
+                    root = cloneDeep(instance);
+                    root.type = `${name}${index}`;
+                }
 
-                    // Expand entity to include possible property names
-                    for (const property in propertyName.value) {
-                        const newInfo: Partial<EntityInfo> = Object.assign({}, info);
-                        newInfo.property = property;
-                        infos.push(newInfo);
-                    }
-                } else {
-                    if (op && name == this.propertyNameKey) {
-                        for (const property in val) {
-                            const newInfo: Partial<EntityInfo> = Object.assign({}, info);
-                            newInfo.property = property;
-                            infos.push(newInfo);
+                if (entityName) {
+                    this.expandEntity(
+                        entityName,
+                        entity as Record<string, unknown[]>,
+                        instance,
+                        root,
+                        op,
+                        property,
+                        turn,
+                        text,
+                        entityToInfo
+                    );
+                } else if (typeof entity === 'object' && entity !== null) {
+                    if (isEmpty(entity)) {
+                        if (isOp) {
+                            // Handle operator with no children.
+                            this.expandEntity(op, null, instance, root, op, property, turn, text, entityToInfo);
+                        } else if (isProperty) {
+                            // Handle property with no children.
+                            this.expandEntity(property, null, instance, root, op, property, turn, text, entityToInfo);
                         }
                     } else {
-                        infos.push(info);
+                        this.expandEntityObject(
+                            entity as Record<string, unknown[]>,
+                            op,
+                            property,
+                            root,
+                            operations,
+                            properties,
+                            turn,
+                            text,
+                            entityToInfo
+                        );
                     }
+                } else if (isOp) {
+                    // Handle global operator with no children in model.
+                    this.expandEntity(op, null, instance, root, op, property, turn, text, entityToInfo);
                 }
-            }
+            });
+        }
+    }
+
+    private stripProperty(name: string): string {
+        return name.endsWith(this.propertyEnding) ? name.substring(0, name.length - this.propertyEnding.length) : name;
+    }
+
+    private expandEntity(
+        name: string,
+        value: Record<string, unknown>,
```

Comment:
[Here](https://github.com/microsoft/botbuilder-js/pull/3628/files#diff-a116dc3a22cc3932c498b73b60c065e4a035945b35eff59dc6130a086739015eR1116) this method is invoked with `value` set to `null`, `value: Record<string, unknown>` disallows `null` (`strict` mode would catch this, but we are in limbo at the moment). Add `| null` to this type just to be explicit.